src/delta.rs

```diff
@@ -458,25 +465,35 @@ fn handle_hunk_header_line(
                 Some(false),
             );
             painter.output_buffer.pop(); // trim newline
-            draw_fn(
-                painter.writer,
-                &painter.output_buffer,
-                &painter.output_buffer,
-                &config.decorations_width,
-                config.hunk_header_style,
-                decoration_ansi_term_style,
-            )?;
-            if !config.hunk_header_style.is_raw {
-                painter.output_buffer.clear()
-            };
+            if !painter.output_buffer.is_empty() {
+                draw_fn(
+                    painter.writer,
+                    &painter.output_buffer,
+                    &painter.output_buffer,
+                    &config.decorations_width,
+                    config.hunk_header_style,
+                    decoration_ansi_term_style,
+                )?;
+            }
```

Comment:
I found draw::write_*** is drawing empty line if the text is empty.
This part is gather text after paint_lines formatted, then empty line should removed because it causes difference error of diff.interactive.diffFilter.

I confirm there is no any case that text will be empty except when color_only.

Reply:
Can you show me how to reproduce the bug/problem you're describing here?

Reply:
@dandavison 
Yes, I made branch `repro_drawfn_bug_for_fix_diff_interactivefilter` in ryuta69/delta to reproduce the bug.

https://github.com/ryuta69/delta/pull/1

You can reproduce the bug by switching to this branch and run `cargo test tests::test_example_diffs::tests::test_diff_interactive_filter_has_same_output_with_corresponding_input -- --nocapture`.

I attached the picture in this PR to show how the bug is caused.

Reply:
We won't need this fix anymore, as same as https://github.com/dandavison/delta/pull/323#r490649549

Reply:
Fixed here 8a5fb65